packages/framework/src/Model/Customer/User.php

```diff
@@ -166,15 +169,31 @@ public function changeEmail(string $email, ?self $userByEmail)
     {
         $email = mb_strtolower($email);
 
-        if ($userByEmail instanceof self) {
-            if (mb_strtolower($userByEmail->getEmail()) === $email && $this !== $userByEmail) {
-                throw new \Shopsys\FrameworkBundle\Model\Customer\Exception\DuplicateEmailException($email);
-            }
+        if ($this !== $userByEmail) {
+            $this->checkDuplicateEmail($email, $this->domainId, $userByEmail);
         }
 
         $this->email = $email;
     }
 
+    /**
+     * @param string $email
+     * @param int $domainId
+     * @param self|null $userByEmail
+     */
+    protected function checkDuplicateEmail(string $email, int $domainId, ?self $userByEmail): void
+    {
+        if ($userByEmail === null) {
+            return;
+        }
+
+        $isSameEmail = (mb_strtolower($userByEmail->getEmail()) === mb_strtolower($email));
```

Comment:
getEmail() should already have a lowercased email
```suggestion
        $isSameEmail = ($userByEmail->getEmail() === mb_strtolower($email));
```